src/libcore/iter/mod.rs

```diff
@@ -683,6 +777,26 @@ impl<A, B> Iterator for Chain<A, B> where
         }
     }
 
+    fn try_fold<Acc, F, R>(&mut self, init: Acc, mut f: F) -> R where
+        Self: Sized, F: FnMut(Acc, Self::Item) -> R, R: Try<Ok=Acc>
+    {
+        match self.state {
```

Comment:
In this method I'm wondering about the benefits of avoiding duplicating expressions that are big loops (`self.a.try_fold(init, f)` and similar for b). I imagine if these are small enough to inline, they are both inlined and duplicated.

Reply:
Oh, binary size.  Makes sense; I can share the calls for the states.

Reply:
code size is relevant for the package of cpu-cache-memory's performance too, just to give a broader sense for what binary size means.